pkg/provisioner/longhornv1.go

```diff
@@ -0,0 +1,457 @@
+			diskv1.DeviceMounted.SetError(p.device, "", err)
+			diskv1.DeviceMounted.SetStatusBool(p.device, false)
+		}
+		return formatted, requeue, err
+	}
+	formatted = true
+	return formatted, false, nil
+}
+
+func (p *LonghornV1Provisioner) UnFormat() (bool, error) {
+	logrus.Infof("%s unformatting Longhorn block device %s", p.name, p.device.Name)
+	return false, nil
+}
+
+func (p *LonghornV1Provisioner) updateDeviceMount(device *diskv1.BlockDevice, devPath string, filesystem *block.FileSystemInfo, needMountUpdate NeedMountUpdateOP) error {
+	logrus.Infof("Prepare to try %s", convertMountStr(needMountUpdate))
+	if device.Status.DeviceStatus.Partitioned {
+		return fmt.Errorf("partitioned device is not supported, please use raw block device instead")
+	}
+	if needMountUpdate.Has(NeedMountUpdateUnmount) {
+		logrus.Infof("Unmount device %s from path %s", device.Name, filesystem.MountPoint)
+		if err := utils.UmountDisk(filesystem.MountPoint); err != nil {
+			return err
+		}
+		diskv1.DeviceMounted.SetError(device, "", nil)
+		diskv1.DeviceMounted.SetStatusBool(device, false)
+	}
+	if needMountUpdate.Has(NeedMountUpdateMount) {
+		expectedMountPoint := extraDiskMountPoint(device)
+		logrus.Infof("Mount deivce %s to %s", device.Name, expectedMountPoint)
+		if err := utils.MountDisk(devPath, expectedMountPoint); err != nil {
+			if utils.IsFSCorrupted(err) {
+				logrus.Errorf("Target device may be corrupted, update FS info.")
+				device.Status.DeviceStatus.FileSystem.Corrupted = true
+				device.Spec.FileSystem.Repaired = false
+			}
+			return err
+		}
+		diskv1.DeviceMounted.SetError(device, "", nil)
+		diskv1.DeviceMounted.SetStatusBool(device, true)
+	}
+	device.Status.DeviceStatus.FileSystem.Corrupted = false
+	return p.updateDeviceFileSystem(device, devPath)
+}
+
+func (p *LonghornV1Provisioner) needFormat() bool {
+	return p.device.Spec.FileSystem.ForceFormatted &&
+		(p.device.Status.DeviceStatus.FileSystem.Corrupted || p.device.Status.DeviceStatus.FileSystem.LastFormattedAt == nil)
+}
+
+// forceFormat simply formats the device to ext4 filesystem
+//
+// - umount the block device if it is mounted
+// - create ext4 filesystem on the block device
+func (p *LonghornV1Provisioner) forceFormatFS(device *diskv1.BlockDevice, devPath string, filesystem *block.FileSystemInfo) (bool, error) {
+	if !p.semaphoreObj.acquire() {
+		logrus.Infof("Hit maximum concurrent count. Requeue device %s", device.Name)
+		return true, nil
+	}
+
+	defer p.semaphoreObj.release()
+
+	// before format, we need to unmount the device if it is mounted
+	if filesystem != nil && filesystem.MountPoint != "" {
+		logrus.Infof("unmount %s for %s", filesystem.MountPoint, device.Name)
+		if err := utils.UmountDisk(filesystem.MountPoint); err != nil {
+			return false, err
+		}
+	}
+
+	// ***TODO***: we should let people to use ext4 or xfs, but now...
+	// make ext4 filesystem format of the partition disk
+	logrus.Debugf("make ext4 filesystem format of device %s", device.Name)
```

Comment:
I don't want to appear small-minded. I think the beginning of a sentence should start with a capital letter.
This should be adapted for all code changes. I will therefore spare myself any further comments in this regard.
```suggestion
	logrus.Debugf("Make ext4 filesystem format of device %s", device.Name)
```

Reply:
I suspect this is form is probably habit related to error strings being uncapitalized. In this case though it's a log message (not an error string) so I agree it should be a proper sentence.

Reply:
Thanks, I correct the above logs. For others, we could do it on another PR.